internal/management/controller/slots/reconciler/replicationslot.go

```diff
@@ -40,13 +40,15 @@ func ReconcileReplicationSlots(
 		return reconcile.Result{}, nil
 	}
 
+	isPrimary := cluster.Status.CurrentPrimary == instanceName || cluster.Status.TargetPrimary == instanceName
+
 	// if the replication slots feature was deactivated, ensure any existing
 	// replication slots get cleaned up
 	if !cluster.Spec.ReplicationSlots.HighAvailability.GetEnabled() {
```

Comment:
this should be !cluster.Spec.ReplicationSlots.GetEnabled()

Reply:
if both HA and user-defined slot are disabled:
  - user defined slot will be delete from standby in reconcling loop
  - HA slot will be deleted from both primary/standby in reconciling loop too.
 
if only HA slot is disabled
  - HA slot will be delete from primary/standby by reconcile loop
  - user defined slot will not be deleted from standby
  
if only user defined slot is been disabled
  - user defined slot will be delete from standby by runnable